metricflow/test/model/validations/test_data_warehouse_tasks.py

```diff
@@ -100,3 +138,76 @@ def test_validate_data_sources(sql_client: SqlClient) -> None:  # noqa: D
     assert len(issues) == 1
     assert issues[0].level == ValidationIssueLevel.ERROR
     assert "Unable to access data source `test_data_source2`" in issues[0].message
+
+
+def test_build_metric_tasks(basic_model: UserConfiguredModel, mf_engine: MetricFlowEngine) -> None:  # noqa: D
+    tasks = DataWarehouseTaskBuilder.gen_metric_tasks(model=basic_model, mf_engine=mf_engine)
+    assert len(tasks) == 1
+    assert tasks[0].query_string == (
```

Comment:
The query text may change, so to make updating tests easier, can you take a look at `assert_snapshot_text_equal` to do the comparison?